wazo_dird/plugins/phonebook_service/plugin.py

```diff
@@ -161,19 +165,33 @@ def import_contacts(
         visible_tenants: list[str],
         phonebook_key: PhonebookKey,
         contacts: list[dict],
-    ) -> tuple[list[ContactInfo], list[dict]]:
+    ) -> tuple[list[ContactInfo], list[ContactEntryError]]:
+        logger.debug(
+            'Processing import of %d contacts in phonebook %s',
+            len(contacts),
+            phonebook_key,
+        )
         to_add, errors = [], []
-        for contact in contacts:
+        for i, contact in enumerate(contacts):
             try:
-                to_add.append(self._validate_contact(contact))
-            except InvalidContactException:
-                errors.append(contact)
+                to_add.append((i, self._validate_contact(contact)))
+            except InvalidContactException as ex:
+                errors.append(
+                    ContactEntryError(
+                        contact=contact,
+                        message=str(ex),
+                        index=i,
+                    )
+                )
+        if errors:
+            return [], errors
 
         created, failed = self._contact_crud.create_many(
-            visible_tenants, phonebook_key, to_add
+            visible_tenants, phonebook_key, [contact for _, contact in to_add]
         )
 
-        return created, failed + errors
+        errors += failed
+        return created, errors
```

Comment:
At this point, `errors` has to be empty (it would have been returned l.187 otherwise)
```suggestion
        return created, failed
```

Reply:
Yes. Since we're not changing the API interface yet(which still returns `{"created": ..., "failed": ...}`), we're keeping the same interface at this level.

Reply:
This comment is not about changing the API interface, just skipping the line `errors += failed` and returning `failed` directly instead, because at that point `errors` should always be empty.